rfcs/active/0000-badge.md

```diff
@@ -0,0 +1,362 @@
+</oui-badge.ribbon>
+```
+
+# Motivation
+
+Badges are a fundamental UI pattern found in many digital interfaces across various platforms. They serve several key purposes:
+
+1. **Information density** - Badges allow developers to display additional information (counts, status, labels) in a compact space.
+2. **Visual indication** - They provide immediate visual cues to users about state, notifications, or important updates.
+3. **Context enhancement** - They enhance other components by providing supplementary information (counters, status indicators).
+
+The OpenUI Design System aims to provide a comprehensive set of common UI components, and badges represent a widely-used pattern that fits within this mission. Adding a standardized badge component would reduce duplicative effort in rebuilding this common UI pattern across different applications and frameworks.
+
+# Detailed Proposal
+
+The `oui-badge` component will provide a versatile, accessible, and themeable implementation of the badge pattern.
+
+## API
+
+### Properties
+
+| Property | Type | Default | Description |
+|----------|------|---------|-------------|
+| `variant` | `'primary' \| 'secondary' \| 'success' \| 'danger' \| 'warning' \| 'info'` | `'primary'` | Determines the visual style of the badge |
+| `shape` | `'default' \| 'pill'` | `'default'` | Controls the border-radius of the badge |
+| `position` | `null \| 'top-start' \| 'top-end' \| 'bottom-start' \| 'bottom-end'` | `null` | When used with a parent element, positions the badge accordingly |
+| `visible` | `boolean` | `true` | Controls whether the badge is displayed |
+| `aria-label` | `string` | `null` | Accessible label (required when badge only contains numbers or non-descriptive content) |
+| `color` | `string` | `null` | Custom color for the badge (hex, rgb, or named color) |
+| `count` | `number \| string` | `null` | Number or content to show in the badge |
+| `overflow-count` | `number` | `99` | Maximum count to show before using the overflow indicator (e.g., "99+") |
+| `dot` | `boolean` | `false` | Whether to display a small dot instead of content |
+| `offset` | `[number, number]` | `[0, 0]` | Offset of the badge [x, y] in pixels |
+| `size` | `'default' \| 'small'` | `'default'` | Size of the badge |
+| `status` | `'success' \| 'processing' \| 'default' \| 'error' \| 'warning'` | `null` | Status indicator type |
+| `show-zero` | `boolean` | `false` | Whether to show the badge when count is zero |
+| `clickable` | `boolean` | `false` | Whether the badge is clickable |
+| `title` | `string` | `null` | Text to show when hovering over the badge |
+
+### Slots
+
+| Slot | Description |
+|------|-------------|
+| `default` | The content of the badge (text or markup) |
+| `count` | Alternative slot for badge content/count |
+
+### Events
+
+| Event | Description |
+|-------|-------------|
+| `click` | Fired when the badge is clicked (if `clickable` is true) |
```

Comment:
This is a little confusing as any element can have a click handler. What does clickable false do? (pointer-events: none?)